synapse/config/key.py

```diff
@@ -194,6 +244,99 @@ def generate_files(self, config):
                     NACL_ED25519, key_id, signing_keys.split("\n")[0]
                 )
                 with open(signing_key_path, "w") as signing_key_file:
-                    write_signing_keys(
-                        signing_key_file, (key,),
+                    write_signing_keys(signing_key_file, (key,))
+
+
+def _perspectives_to_key_servers(config):
+    """Convert old-style 'perspectives' configs into new-style 'trusted_key_servers'
+
+    Returns an iterable of entries to add to trusted_key_servers.
+    """
+
+    # 'perspectives' looks like:
+    #
+    # perspectives:
+    #   servers:
+    #     "matrix.org":
+    #       verify_keys:
+    #         "ed25519:auto":
+    #           key: "Noi6WqcDj0QmPxCNQqgezwTlBKrfqehY1u2FyWP9uYw"
+    #
+    # the equivalent trusted_keys entry looks like:
+    #
+    # {
+    #     "server_name": "matrix.org",
+    #     "verify_keys": {
+    #         "ed25519:auto": "Noi6WqcDj0QmPxCNQqgezwTlBKrfqehY1u2FyWP9uYw",
+    #     }
+    # }
```

Comment:
(I find it a bit confusing that they you're comparing a) different levels of dict and b) using yaml vs json)

Reply:
fair